### PR TITLE
Dataset trigger

### PR DESCRIPTION
**Pull request by richhuth**

### CODE REVIEW

File: langwatch/src/components/AddTriggerDrawer.tsx [hunk 1]
```diff
@@ -59,24 +78,117 @@ export function TriggerDrawer() {
...
+      ...filterParams,
+      projectId: project?.id ?? "",
+    },
+    {
+      enabled: !!project,
+      refetchOnWindowFocus: false,
+    }
+  );
+
+  const columnDefs = useMemo(() => {
```
Comment (rogeriochaves) [inline langwatch/src/components/AddTriggerDrawer.tsx]: is this something that could go inside the DatasetMappingPreview? This way we wouldn't need to duplicate the definitions. We already need to send the columnTypes down anyway

File: langwatch/src/components/datasets/DatasetMappingPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+export function DatasetMappingPreview({
+  traces,
+  columnTypes,
+  columnDefs,
+  rowData,
+  onEditColumns,
+  onRowDataChange,
+  paragraph,
+}: DatasetMappingPreviewProps) {
+  console.log("rowData", rowData);
```
Comment (rogeriochaves) [inline langwatch/src/components/datasets/DatasetMappingPreview.tsx]: forgotten console log

File: langwatch/src/pages/api/triggers/index.ts [hunk 3]
```diff
@@ -130,7 +147,7 @@ const getTracesForAlert = async (trigger: Trigger, projects: Project[]) => {
 
     if (action === TriggerAction.SEND_EMAIL) {
       triggerInfo = {
-        triggerEmails: (actionParams as ActionParams)?.members ?? [],
+        triggerEmails: (actionParams as unknown as ActionParams)?.members ?? [],
```
Comment (rogeriochaves) [inline langwatch/src/pages/api/triggers/index.ts:150]: hmm why was it necessary to add unknown here now?

File: langwatch/src/components/datasets/DatasetMapping.tsx [hunk 4]
```diff
@@ -507,29 +508,11 @@ export const TracesMapping = ({
     }
 
     for (const trace of expandedTraces) {
-      const entry = Object.fromEntries(
```
Comment (rogeriochaves) [inline langwatch/src/components/datasets/DatasetMapping.tsx:510]: how come this piece was not moved along?

```
    for (const expansion of expansions) {
      expandedTraces = expandedTraces.flatMap((trace) =>
        TRACE_EXPANSIONS[expansion].expansion(trace)
      );
```

you are just passing `expansions` as a set but never really expanding it as the above for loop when the trigger is triggered

please move this and the whole for-loop along, and do a proper testing of an add do dataset trigger with expansions

File: langwatch/src/pages/api/triggers/index.ts (hunk 3 above)
Comment (richhuth) [reply langwatch/src/pages/api/triggers/index.ts:150]: so it brought up a typscripts error, and it suggested this, to went with it

**Pull request merged**